File: src/main/java/pro/evanwright/saphira/exception/DatabaseClientInitializationException.java
package pro.evanwright.saphira.exception;

import pro.evanwright.saphira.DatabaseClient;

/**
 * A wrapper around all exceptions that can occur when a {@link DatabaseClient} initializes.
 */
public class DatabaseClientInitializationException extends RuntimeException {
    public DatabaseClientInitializationException(String errorMessage, Throwable throwable) {
        super(errorMessage, throwable);
    }

    public DatabaseClientInitializationException(String errorMessage) {
        super(errorMessage);
    }
}

File: src/main/java/pro/evanwright/saphira/exception/DatabaseClientInitializationException.kt
package pro.evanwright.saphira.exception

/**
 * An unchecked, wrapper [Exception] around all exceptions that can occur
 * when a [DatabaseClient] initializes.
 */
class DatabaseClientInitializationException : RuntimeException {
    constructor(throwable: Throwable?) : super(throwable) {}
    constructor(errorMessage: String?, throwable: Throwable?) : super(errorMessage, throwable) {}
}